openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -14,1043 +14,864 @@
-            // Create array of primary indicators
-            pdTempCategoryIndicator.push(pdData.name);
-            tempCatWeight[pdData.name.toLowerCase()] = (pdData.weight);
-            // Create a key pair for each category indicator and its respective child level
-            pdTempPILevel[pdData.name.toLowerCase()] = (pdData.children[0].level);
+    // process each Social Vulnerability Index nodes
+    // Get all the primary indicators
+    var ct = 0;
+
+    for (var i = 0; i < socialVulnIndex.length; i++) {
+        for (var e = 0 ; e < socialVulnIndex[i].children.length; e++, ct++ ) {
+            allPrimaryIndicators.push(socialVulnIndex[i].children[e].field);
         }
-        (pdData.children || []).forEach(function(currentItem) {
-            findCategoryIndicators(currentItem, [ci]);
-        });
     }
 
-    function checkElementCategory(elementName) {
-        var foo = pdTempPrimaryIndicatorLevels[elementName];
+    /////////////////////////////////////////
+    //// Compute the Category indicators ////
+    /////////////////////////////////////////
 
-        for(var prop in pdTempPILevel) {
-            if(pdTempPILevel.hasOwnProperty(prop)) {
-                if(pdTempPILevel[prop] === foo) {
-                    return prop;
-                }
+    //build the category indicator object
+    var catData = [];
+    tempString = [];
+
+    function generateObject(temp) {
+        var munic = temp[0];
+        var theme = temp[1];
+        var value = parseFloat(temp[2]);
+        // add the theme and value to each category data object
+        for (var i = 0; i < catData.length; i++) {
+            if (catData[i].region == munic) {
+                catData[i][theme] = value;
             }
         }
     }
 
-    function findSvi(pdData, svi) {
-        // Find all of the primary indicators
-        if (svi.some(function(currentValue) {
-            return (pdData.name == currentValue);
-        })) {
-            tempSviWeight = pdData.weight;
-        }
-        (pdData.children || []).forEach(function(currentItem) {
-            findSvi(currentItem, [svi]);
-        });
+    // setup catData with all the municipalities
+    var la = layerAttributes.features;
+    for (var s = 0; s < la.length; s++) {
+        var temp = {};
+        temp.region = la[s].properties[selectedRegion];
+        catData.push(temp);
     }
 
-    function findAalWeight(pdData, aal) {
-        // Find the aal value
-        if (aal.some(function(currentValue) {
-            return (pdData.name == currentValue);
-        })) {
-            tempAalWeight = pdData.weight;
+    // Find the theme information
+    for (var m = 0; m < socialVulnIndex.length; m++) {
+        var operator = socialVulnIndex[m].operator;
+        var weight = socialVulnIndex[m].weight;
+        var name = socialVulnIndex[m].name;
+        allSVIThemes.push(name);
+        var tempChildren = socialVulnIndex[m].children;
+        var indicatorChildrenKey = [];
```

Comment:
indicatorChildrenKey was also defined outside. Please check the scopes. I also suggest to refactor the name into indicatorChildrenKeys